docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6328,409 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+consists of unions which are similar to actions. One use of `GenericTable` in a
+server backend is with TDI [Table Driven
+Interface](https://github.com/p4lang/tdi), but targets can use Generic table to
+map to their own specific feature implementations as well.
+
+## Generic Table ID Space { #sec-generic-table-id-space }
+
+A field in the `GenericTable` message in the P4info defines a
+`generic_table_type_id`.
+
+It is an 8-bit unsigned integer which uniquely identifies
+the generic_table_type in the context of the architecture.
+This value is in the range of `[0x00, 0xff]`.
+The ID in the preamble is created as `(0x18 << 24) + (generic_table_type_id
+<< 16) + (16-bit ID generated by the P4 compiler)`. The `generic_table_type`
+depicts a new table "type" which would have similar properties.
+
+Let's say we have 2 target specific P4 externs called `XyzCounter` and
+`XyzMeter` and each of them has 2 instances. `XyzCounter` has
+`generic_table_type` as 1 and `XyzMeter` as 2.
+
+```text
+XyzCounter1 : 0x18 01 00 01
+XyzCounter2 : 0x18 01 00 02
+
+XyzMeter1 : 0x18 02 00 01
+XyzMeter2 : 0x18 02 00 02
+```
+The above system ensures that the instance space allows for 65355
+instances. The table type space is limited to 255.
+
+Some optional guidelines:
+* For P4 externs, any new extern to have a new table type.
+* For target-specific fixed function blocks, options are
+  * All fixed tables grouped under one giant non_p4 table type
+  * Tables divided on the basis of categories. Categories are defined in the
+    spec but aren't part of the P4info.
+  * Every table has its own table-type. This can however lead to running out
+    of the 8 bit type ID space soon.
+Note that this value does not need
+to be unique across all architectures from all organizations, since at any
+given time every device managed by a P4Runtime server maps to a single P4Info
+message and a single architecture.
+
+
+## GenericTable P4info { #sec-p4info-generic-table}
+
+`GenericTable` message in P4info defines a Generic Table. It contains several
+instances in `GenericTableInstance` rather than duplicating it every time for a
+new instance just like in `Extern`.
+It defines the following fields
+
+* `type_id`, an 8-bit unsigned integer which uniquely identifies
+  the generic_table_type in the context of the architecture. More details
+  in the [Generic Table ID space](#sec-generic-table-id-space) section
+
+* `type_name`, which specifies the "type" of the `GenericTable`. For example,
+  if the target has a meter called `SpecialMeter`, then this would contain
+  "SpecialMeter". In cases of non-P4 entities, it is a "type" which helps
+  identify this entity with a feature for example "MulticastGroup". For non-
+  P4 entities, sometimes the instance name and type name are the same
+
+* `repeated string properties`, please check generic-table
+  properties under section [GenericTable categories]
+  (#sec-generic-table-categories) for more details.
```

Comment:
I am not sure, but perhaps Madoko markup syntax rules require [GenericTable categories](#sec-generic-table-categories) to have no white space after `]` and before `(`.

You should try generating PDF and HTML output from the Madoko file to check not just this, but the entire new version of the document to see if there are any other Madoko markup issues.  I don't see precise instructions for installing all required software on a Linux system to build the P4Runtime API spec from source code yet, so I will create a PR to add some instructions.